src/test/java/core/basesyntax/service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,118 @@
+package core.basesyntax.service;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceTest {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_USER_AGE = 18;
+    private static RegistrationService registrationService;
+    private User validUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        validUser = new User();
+        validUser.setAge(18);
+        validUser.setLogin("valLog");
+        validUser.setPassword("valPas");
+    }
+
+    @Test
+    void test_register_validUser_ok() {
+        User actualUser = registrationService.register(validUser);
+        User expectUser = validUser;
+        Assertions.assertEquals(expectUser, actualUser);
+    }
+
+    @Test
+    void test_register_existUser_notOk() {
+        String login = "ValidLogin";
+        User sameLoginUser = new User();
+        sameLoginUser.setLogin(login);
+        validUser.setLogin(login);
+        Storage.people.add(sameLoginUser);
+        Assertions.assertThrows(RegistrationException.class,
+                () -> registrationService.register(validUser),
+                "There is shouldn't add a user with already exist login in the storage. Login: "
+                + login);
+    }
+
+    @Test
+    void test_register_nullUser_notOk() {
+        Assertions.assertThrows(RegistrationException.class,
+                () -> registrationService.register(null),
+                "There is shouldn't to pass null User");
+    }
+
+    @Test
+    void test_register_nullLogin_notOk() {
+        User nullLoginUser = validUser;
```

Comment:
In my opinion this variable is redundant you can set invalid value directly to user object

Reply:
Yes, mostly, it is only to name local variable

Reply:
Just rename to `user` and use it ewerywhere.